File: Share/src/main/java/com/buguagaoshu/share/domain/Tag.java
package com.buguagaoshu.share.domain;

import lombok.Data;

import javax.persistence.*;

/**
 * @author Pu Zhiwei {@literal puzhiweipuzhiwei@foxmail.com}
 * create          2021-07-30 23:01
 */
@Data
@Entity
@Table(name = "tag_table")
public class Tag {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String title;

    private Boolean lock;

    private String password;
}

File: Go/internal/model/tag.go
package model

// Tag 对齐 Spring com.buguagaoshu.share.domain.Tag（表 tag_table）
type Tag struct {
	ID       *int64 `gorm:"primaryKey;autoIncrement;column:id" json:"id"`
	Title    string `gorm:"column:title" json:"title"`
	Lock     *bool  `gorm:"column:lock" json:"lock"`
	Password string `gorm:"column:password" json:"password"`
}

func (Tag) TableName() string { return "tag_table" }
